src/stream/futures.rs

```diff
@@ -0,0 +1,71 @@
+use future::{Future, IntoFuture};
+use stream::Stream;
+use poll::Poll;
+use Async;
+
+use std::prelude::v1::*;
+
+/// An adaptor for a stream of futures to execute the futures concurrently, if
+/// possible, delivering results as they become available.
+///
+/// This adaptor will return their results in the order that they complete.
+/// This is created by the `futures` method.
+///
+#[must_use = "streams do nothing unless polled"]
+pub struct Futures<F>
+    where F: Future
+{
+    futures: Vec<Option<F>>,
+    active: usize,
+}
+
+/// Converts a `Vec` of `IntoFuture` elements into a `Stream` over the underlying
+/// futures. The created stream will yield into values produced by those futures.
+/// Similar to `buffer_unordered` the order of the produced items will not be equal
+/// with the order of elements in the input vector.
+///
+pub fn futures<F>(futures: Vec<F>) -> Futures<F::Future>
+    where F: IntoFuture
+{
+    Futures {
+        active: futures.len(),
+        futures: futures.into_iter().map(IntoFuture::into_future).map(Some).collect(),
+    }
+}
+
+
+impl<F> Stream for Futures<F>
+    where F: Future
+{
+    type Item = F::Item;
+    type Error = F::Error;
+
+    fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
+        if self.active == 0 {
+            return Ok(Async::Ready(None));
+        }
+        for f in self.futures.iter_mut() {
+            let ret;
+            match *f {
+                None => continue,
+                Some(ref mut future) => {
+                    match future.poll() {
+                        Ok(Async::NotReady) => {
+                            continue;
+                        }
+                        Ok(Async::Ready(e)) => {
+                            ret = e;
```

Comment:
Just yield `e` (`=> e,`). However, `e` usually refers to errors so I'd use `v` or `val`.